web/modules/custom/brms_sor/src/Service/FmsSorApi.php

```diff
@@ -0,0 +1,252 @@
+<?php declare(strict_types = 1);
+
+namespace Drupal\brms_sor\Service;
+
+use Drupal\Component\Datetime\TimeInterface;
+use Drupal\Component\Serialization\Json;
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+use Drupal\Core\Site\Settings;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\key\KeyRepositoryInterface;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Http\Client\ClientInterface;
+
+/**
+ * Implementation of System of Record API interface for FMS.
+ */
+final class FmsSorApi implements SorApiInterface {
+  use StringTranslationTrait;
+
+  /**
+   * GuzzleHttp client.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * The logger factory.
+   *
+   * @var \Drupal\Core\Logger\LoggerChannelFactoryInterface
+   */
+  protected $logger;
+
+  /**
+   * The JSON serializer service.
+   *
+   * @var \Drupal\Component\Serialization\Json
+   */
+  protected $json;
+
+  /**
+   * The cache backend.
+   *
+   * @var \Drupal\Core\Cache\CacheBackendInterface
+   */
+  protected $cache;
+
+  /**
+   * FMS API auth URL.
+   *
+   * @var string
+   */
+  protected $authUrl;
+
+  /**
+   * FMS API base URL.
+   *
+   * @var string
+   */
+  protected $baseUrl;
+
+  /**
+   * FMS API client id.
+   *
+   * @var string
+   */
+  protected $clientId;
+
+  /**
+   * Secret of the DnB client.
+   *
+   * @var string
+   */
+  protected $clientSecret;
+
+  /**
+   * Constructs a FmsSorApi object.
+   *
+   * @param \GuzzleHttp\ClientInterface $http_client
+   *   The GuzzleHttp Client.
+   * @param \Drupal\Core\Logger\LoggerChannelFactoryInterface $logger_factory
+   *   The logger factory.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cache_backend
+   *   The cache backend.
+   * @param \Drupal\key\KeyRepositoryInterface $key_repository
+   *   The key repository.
+   * @param \Drupal\Component\Serialization\Json $json_service
+   *   The JSON serializer service.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   */
+  public function __construct(
+    protected ClientInterface $http_client,
+    protected LoggerChannelFactoryInterface $logger_factory,
+    protected CacheBackendInterface $cache_backend,
+    protected KeyRepositoryInterface $key_repository,
+    protected Json $json_service,
+    protected TimeInterface $time,
+  ) {
+    $this->httpClient = $http_client;
+    $this->logger = $logger_factory;
+    $this->cache = $cache_backend;
+    $this->json = $json_service;
+    $settings = Settings::get('brms_sor');
+    $this->authUrl = $settings['fms_api_auth_url'];
+    $this->baseUrl = $settings['fms_api_base_url'];
+    $this->clientId = $settings['fms_api_client_id'];
+    $this->clientSecret = $key_repository->getKey('fms_api')->getKeyValue();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getAccessToken(bool $use_cached_token = TRUE): string {
```

Comment:
Why is it a decision? shouldn't this be if the token is available in cache, use that?

Reply:
removed the use_cached_token option